### ISSUE TITLE
Fifo bug on second read

### ISSUE THREAD

**Issue by eerimoq** [jonasblixt/bpak#2]
The second read reads zero bytes, but 16 are available.

``` c
TEST(fifo_erik)
{
    int rc;
    struct bpak_io *f = NULL;
    struct bpak_io_fifo *fifo_p;
    char bfr[17];
    size_t bytes = 0;

    memset(bfr, 0, sizeof(bfr));

    rc = bpak_io_fifo_init(&f, 16);
    ASSERT_EQ(rc, BPAK_OK);

    /* First write and read. */
    bytes = bpak_io_write(f, "1234567890123456", 16);
    ASSERT_EQ(bytes, 16);

    bytes = bpak_io_read(f, bfr, 16);
    ASSERT_EQ(bytes, 16);
    ASSERT_EQ((char *) bfr, "1234567890123456");

    /* The same write and read again. Now read fails. */
    bytes = bpak_io_write(f, "1234567890123456", 16);
    ASSERT_EQ(bytes, 16);

    bytes = bpak_io_read(f, bfr, 16);          // This read returns 0, but should return 16!
    ASSERT_EQ(bytes, 16);
    ASSERT_EQ((char *) bfr, "1234567890123456");

    rc = bpak_io_close(f);
    ASSERT_EQ(rc, BPAK_OK);
}
```

**Comment by jonasblixt** [comment]
Thanks for the bug report. Fixed in v0.3.4

**Comment by eerimoq** [comment]
Thanks for the fast fix and release! =)
